pkg/controller/ec2/securitygroup/controller.go

```diff
@@ -185,18 +186,29 @@ func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.Ex
 		VpcId:       cr.Spec.ForProvider.VPCID,
 		Description: aws.String(cr.Spec.ForProvider.Description),
 	}).Send(ctx)
-
 	if err != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
 	}
-
 	if result.CreateSecurityGroupOutput.GroupId == nil {
 		return managed.ExternalCreation{}, errors.New(errCreate)
 	}
-
 	meta.SetExternalName(cr, aws.StringValue(result.GroupId))
 
-	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errSpecUpdate)
+	if err := e.kube.Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errSpecUpdate)
+	}
+	// NOTE(muvaf): AWS creates an initial egress rule and there is no way to
+	// disable it with the create call. So, we revoke it right after the creation.
```

Comment:
Should we always revoke? If the user supplies no egress rules should we assume they want the default?

Reply:
I think so. The CR is supposed to serve as the source of truth. In other cases, we'd cover that in late-init but when user does give an entry in that array the default one is not late-inited so we get into a loop of up-to-date.

Also in AWS console, that rule is populated in creation options and when you remove it and click create, you see revoke call being made in the progress bar.

Reply:
Nice! This sounds like a reasonable solution here, thanks for explaining :)